### ISSUE TITLE
When sub-projects depend on different versions of a library - only one is shown in the report

### ISSUE THREAD

**Issue by aclement** [pivotal/LicenseFinder#241]
In the spring cloud services project we have a top level repo with multiple sub projects. There is only a build.gradle file in the root (not in the sub projects). When I run the license finder:

```
license_finder report --gradle-command=./gradlew --recursive > report.txt
```

It produces a report that contains this line:

```
tomcat-juli,8.0.36,Apache 2.0
```

When we run that command on the CI pipeline, it produces a report containing this line:

```
tomcat-juli,8.0.15,Apache 2.0
```

Which is right? They are both right - one sub-project depends transitively on 8.0.15, one depends on 8.0.36. Once built the sub-projects run as different executables on different JVMs so both versions will be in use.

The top level report should include both (and possibly indicate which project produced which). The per machine difference seems like it just an ordering thing and may simply depend on which sub-project we go into first/last.

Or, is there an option I don't know about to make it behave already?

Feel free to contact me for repo access.

**Comment by flavorjones** [comment]
Ack-ing. Ryan Morgan has poked me about this, will try to address it soon.

**Comment by kdykeman** [comment]
Currently, the [hash for a Package object](https://github.com/pivotal/LicenseFinder/blob/d6dadfb747a9b95e0548487aad8400f60e64d1d2/lib/license_finder/package.rb#L109-L111) is based solely on `name`, so packages with the same name test as equivalent.  There's a commit coming shortly that's going to add `version` to the hash (and thus equivalency test).  Does anybody have concerns about that change or think other attributes should be included?

**Comment by mainej** [comment]
That will mean that anywhere packages are sorted or uniqued, behavior will
change. It's likely that two versions of a package will no longer be
considered equal and therefore users will have to re-approve whenever they
upgrade packages. Some users have requested that behavior as a feature but
most users seem to prefer the existing behavior where packages stay
approved even as they are upgraded. The tests should tell you whether I'm
right about all this. Anyway, this is a big decision and should be
considered carefully.

**Comment by kdykeman** [comment]
@mainej Thanks for that!  I hadn't been thinking of that use case.  Definitely need to apply further consideration.

**Comment by ryanmattcollins** [comment]
An alternative might be to include the license type in the hash. Arguably any dependencies that have the same name, but different license types are not unique.

**Comment by kdykeman** [comment]
This should be resolved and working in version 3.0.2. It worked in some fashion starting in 3.0.0, but npm wasn't behaving correctly until 3.0.2.
